.NET/Microsoft.Recognizers.Text.DateTime/Spanish/Parsers/SpanishDateParserConfiguration.cs

```diff
@@ -42,7 +42,7 @@ public SpanishDateParserConfiguration(ICommonDateTimeParserConfiguration config)
             UpcomingPrefixRegex = new Regex(DateTimeDefinitions.UpcomingPrefixRegex, RegexFlags);
             PastPrefixRegex = new Regex(DateTimeDefinitions.PastPrefixRegex, RegexFlags);
 
-            DayOfMonth = config.DayOfMonth;
+            DayOfMonth = DateTimeDefinitions.DayOfMonth.ToImmutableDictionary();

```

Comment:
Why this change?

Reply:
config.DayOfMonth gets its values from BaseDateTime.DayOfMonthDictionary which only contains digit numbers. We use instead the Spanish version of the dictionary which contains also written day numbers in order to parse the cases like "on the twenty-six". 
But this can probably be avoided by enabling ForTheRegex in Spanish, instead of using OnRegex to parse these patterns. I will try this solution.